Source/Core/DolphinQt/GCMemcardManager.cpp

```diff
@@ -298,157 +321,309 @@ void GCMemcardManager::SetSlotFileInteractive(int slot)
     m_slot_file_edit[slot]->setText(path);
 }
 
-void GCMemcardManager::ExportFiles(bool prompt)
+std::vector<u8> GCMemcardManager::GetSelectedFileIndices()
 {
-  auto selection = m_slot_table[m_active_slot]->selectedItems();
-  auto& memcard = m_slot_memcard[m_active_slot];
-
-  auto count = selection.count() / m_slot_table[m_active_slot]->columnCount();
-
-  for (int i = 0; i < count; i++)
+  const auto selection = m_slot_table[m_active_slot]->selectedItems();
+  std::vector<bool> lookup(Memcard::DIRLEN, false);
+  for (const auto* item : selection)
   {
-    auto sel = selection[i * m_slot_table[m_active_slot]->columnCount()];
-    int file_index = memcard->GetFileIndex(m_slot_table[m_active_slot]->row(sel));
-
-    std::string gci_filename;
-    if (!memcard->GCI_FileName(file_index, gci_filename))
-      return;
-
-    QString path;
-    if (prompt)
-    {
-      path = QFileDialog::getSaveFileName(
-          this, tr("Export Save File"),
-          QString::fromStdString(File::GetUserPath(D_GCUSER_IDX)) +
-              QStringLiteral("/%1").arg(QString::fromStdString(gci_filename)),
-          tr("Native GCI File (*.gci)") + QStringLiteral(";;") +
-              tr("MadCatz Gameshark files(*.gcs)") + QStringLiteral(";;") +
-              tr("Datel MaxDrive/Pro files(*.sav)"));
-
-      if (path.isEmpty())
-        return;
-    }
-    else
+    const int index = item->data(Qt::UserRole).toInt();
+    if (index < 0 || index >= static_cast<int>(Memcard::DIRLEN))
     {
-      path = QString::fromStdString(File::GetUserPath(D_GCUSER_IDX)) +
-             QStringLiteral("/%1").arg(QString::fromStdString(gci_filename));
+      PanicAlert("Data inconsistency in GCMemcardManager, aborting action.");
```

Comment:
```suggestion
      QMessageBox::warning(tr("Data inconsistency in GCMemcardManager, aborting action."));
```

or use `ModelMessageBox::warning`